src/context/blocked.rs

```diff
@@ -28,8 +187,54 @@ impl Context {
             )
         };
 
-        BlockedClient {
-            inner: blocked_client,
+        BlockedClient::new(blocked_client)
+    }
+
+    /// Blocks a client during authentication and registers callbacks
+    ///
+    /// Wrapper around ValkeyModule_BlockClientOnAuth. Used for asynchronous authentication
+    /// processing.
+    ///
+    /// # Arguments
+    /// * `auth_reply_callback` - Callback executed when authentication completes
+    /// * `free_priv_callback` - Optional callback for cleaning up private data
+    ///
+    /// # Returns
+    /// * `BlockedClient<T>` - Handle to manage the blocked client
+    #[must_use]
+    #[cfg(all(any(
+        feature = "min-redis-compatibility-version-7-2",
+        feature = "min-valkey-compatibility-version-8-0"
+    ),))]
+    pub fn block_client_on_auth<T: 'static + Send + Clone>(
+        &self,
+        auth_reply_callback: fn(
+            &Context,
+            ValkeyString,
+            ValkeyString,
+            T,
+        ) -> Result<c_int, ValkeyError>,
+        free_priv_callback: Option<FreePrivDataCallback<T>>,
+    ) -> BlockedClient<T> {
+        unsafe {
+            let blocked_client = raw::RedisModule_BlockClientOnAuth.unwrap()(
+                self.ctx,
+                Some(auth_reply_wrapper::<T>),
+                Some(free_callback_wrapper::<T>),
+            );
+
+            BlockedClient::with_auth_callback(
+                blocked_client,
+                auth_reply_callback,
+                free_priv_callback,
+            )
         }
     }
+
+    /// Retrieves private data associated with a blocked client
+    ///
+    /// Wrapper around ValkeyModule_GetBlockedClientPrivateData.
+    pub fn get_blocked_client_privdata(&self) -> *mut c_void {
```

Comment:
Instead of returning a raw `* c_void` pointer that contains the SDK's internally managed data structure (currently a tuple), it would be better to return `data: T` just like we do in the `set_blocked_private_data` API